modules/scheduler/gke-cluster/main.tf

```diff
@@ -168,6 +176,14 @@ resource "google_container_cluster" "gke_cluster" {
     ignore_changes = [
       node_config
     ]
+    precondition {
+      condition     = !(!coalesce(var.enable_dataplane_v2, true) && (coalesce(var.enable_multi_networking, false) || length(var.additional_networks) > 0))
```

Comment:
I believe pre-conditions can use local vars. I think better than testing the `var.` here and having coalesce in multiple places, it would be better to test your `local. derived_enable_dataplane_v2` and `local. derived_enable_multi_networking` which will actually be used to define the configuration. This in combination with always taking a user provided option first should make for clear configuration.

Reply:
Here I used the preconditions as a replacement for cross-variables validation. I think here if we validate the input variables instead of processed local variables, we can give more to the point error description of the root problematic variables combinations. This validations also help us not to ignore user explicit actions; explicit settings are either acceptable or rejected here. What do you think?